### ISSUE TITLE
BigDecmial generation fails on Java 11

### ISSUE THREAD

**Issue by iljakorneckis** [Nylle/JavaFixture#85]
When generating BigDecimals on Java 11, JavaFixture populates the **stringCache** field with a random not-a-number string. This affects the `.toString()` method at the very least and, by proxy, affects JSON generated by ObjectMapper. The `.intValue()` also produces zero. On Java 17, the same field is populated by a correct string representation of the decimal.

Could it be that the reason it works on Java17 is the module system preventing access to internals and forcing JF to use a constructor?

#### Test expected to pass
```java
@Test
void bigDecimalTest() {
     var bigDecimal = Fixture.fixture().create(BigDecimal.class);

     assertThat(bigDecimal.intValue()).isNotZero();
     assertThat(new BigDecimal(bigDecimal.toString())).isEqualTo(bigDecimal);
}
```

#### Json generated by Object mapper:
The **cpc** field is a BigDecimal. It's not even valid Json, which is more of a problem with ObjectMapper than JavaFixture tho.
```json
{
 "cpc":EMxJJRhq3XBJVv8vayJ1ivNOzWoO6tuegfY1DBCMLb9MrfYKdJSq19RhKB3COzqEoEOmfsD8naQbKvzTXLFyKmzYu45wV6YiDbTt2X4OimoFZ3hk7fUm4zmdOYK7TuOC,
 "category_id":-1313760193,
}
```

#### Java 11 BigDecimal
![Screenshot from 2023-03-02 11-43-41](https://user-images.githubusercontent.com/7916600/222410659-a5d6c597-040d-4131-831e-770ff9957820.png)

#### Java 17 BigDecimal
![Screenshot from 2023-03-02 11-45-19](https://user-images.githubusercontent.com/7916600/222410741-c9a2f7a2-6e66-404b-8ba0-a620bee1e697.png)

**Comment by iljakorneckis** [comment]
I suspect that the reason it works on Java 17 is the module system preventing access to the internals of BigDecimal.

**Comment by akutschera** [comment]
Yes. That is a bug. We should not rely on reflection for a BigDecimal, similar to a BigInteger.
Therefore I propose we add it to the SpecialSpecimen classes and create one using a random value in a constructor.

**Comment by Nylle** [comment]
Fixed in version 2.9.5
